src/RA_Core.cpp

```diff
@@ -400,7 +428,9 @@ API int CCONV _RA_OnLoadNewRom(const BYTE* pROM, unsigned int nROMSize)
             //	Some other fatal error... panic?
             ASSERT(!"Unknown error from requestgameid.php");
 
-            MessageBox(g_RAMainWnd, NativeStr("Error from " RA_HOST_URL "!\n").c_str(), TEXT("Error returned!"), MB_OK);
+            std::ostringstream oss;
+            oss << "Game not loaded.\nError from " << _RA_HostName() << "!";
+            MessageBox(g_RAMainWnd, oss.str().c_str(), TEXT("Error returned!"), MB_OK | MB_ICONERROR);
```

Comment:
Be careful with the `oss.str().c_str()` part. Sometimes it dangles (appends garbage). I'd suggest using a "`tostringstream`" since `MessageBox `could be `ASCII `or `Unicode`.
like: `using tostringstream = std::basic_string<TCHAR>;`

Reply:
Good catch. I've wrapped it in a `NativeStr` call for Unicode support.

I'm not sure what you mean by "dangles (appends garbage)". `str()` returns a `std::string` object, which should live long enough for the `MessageBox` call. If not, using `NativeStr` isn't going to help because it does the same thing.